### PR TITLE
[aws_iid] tolerate network interfaces not being ordered by device index

### PR DESCRIPTION
**Pull request by Manbeardo**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

`aws_iid` agent NodeAttestor

**Description of change**

This diff updates the `aws_iid` node attestation process to search the list of network interfaces for the interface with device index 0 instead of bailing out if the first element of the list does not have device index 0.

I don't believe this change requires a documentation update. Please point me in the right direction if that's incorrect.

**Which issue this PR fixes**
https://github.com/spiffe/spire/issues/5299

### CODE REVIEW

File: pkg/server/plugin/nodeattestor/awsiid/iid.go [hunk 1]
```diff
@@ -328,13 +329,17 @@ func (p *IIDAttestorPlugin) SetLogger(log hclog.Logger) {
...
-	if ifaceZeroDeviceIndex != 0 {
-		return fmt.Errorf("the EC2 instance network interface attachment device index must be zero (has %d)", ifaceZeroDeviceIndex)
+	ifaceZeroIndex := slices.IndexFunc(
+		instance.NetworkInterfaces,
+		func(net ec2types.InstanceNetworkInterface) bool {
+			return *net.Attachment.DeviceIndex == 0
+		},
+	)
+	if ifaceZeroIndex == -1 {
+		return fmt.Errorf("the EC2 instance network interface with device index 0 is inaccessible")
```
Comment (amartinezfayo) [inline pkg/server/plugin/nodeattestor/awsiid/iid.go:339]: No need for formatting the error.
```suggestion
		return errors.New("the EC2 instance network interface with device index 0 is inaccessible")
```

**Review by amartinezfayo — COMMENTED** [review]
Thank you @Manbeardo for this contribution, it looks great.
Just a small comment.

**Comment by Manbeardo** [comment]
replaced `fmt.Errorf` with `errors.New` where applicable

**Review by amartinezfayo — APPROVED** [review]
Thank you @Manbeardo!

**Pull request merged**